docs/rfc/rfc-10.adoc

```diff
@@ -0,0 +1,248 @@
+placed in `tbtc-v2/solidity` project. The contract will receive tBTC from Solana
+via Wormhole Token Bridge and request redemption in the tBTC `Bridge` contract. 
+
+The contract needs to be upgradeable.
+
+==== Relayer bot
+
+To optimize the user experience, a relayer bot needs to be implemented. The
+relayer's responsibility will be:
+
+- Deliver Wormhole VAA to `SolanaDepositor` contract once the deposit was
+  revealed on Solana.
+- Request bridging tBTC from Ethereum to Solana once the tBTC for the deposit
+  has been optimistically minted or the deposit was swept.
+- Redeem bridged tBTC on Solana to the depositor address once the tBTC minted on
+  L1 has been successfully bridged to Solana.
+- Redeem bridged tBTC on Ethereum to `SolanaRedeemer` contract once the
+  redemption was requested on Solana and tBTC was bridged back via the Wormhole
+  Token Bridge.
+
+==== Optimistic minting changes
+
+Currently, the optimistic minting fee is evaluated at the moment of finalizing
+the mint. The `SolanaDepositor` needs to know how much tBTC should be bridged to
+Solana to the given depositor address. To evaluate the amount, the
+`SolanaDepositor` contract needs to know if the deposit was optimistically
+minted and what was the fee during the mint. The `OptimisticMintingRequest`
+struct has to be enhanced with fee information captured at the moment of
+requesting or finalizing the mint.
+
+==== Deposit flow
+
+From the user's perspective:
+
+1. The user generates a Bitcoin deposit address in the Threshold Dashboard.
+2. The user makes a Bitcoin deposit.
+3. The user reveals their deposit with transaction on Solana in the Threshold
+   Dashboard.
+4. After some time, the user receives their tBTC under the Solana address.
+
+With smart contract interactions:
+
+1. The user generates a Bitcoin deposit address in the Threshold Dashboard.
+2. The user makes a Bitcoin deposit.
+3. The user reveals their deposit with transaction on Solana in the Threshold
+   Dashboard.
+4. `SolanaGateway` contract sends Wormhole VAA from Solana to Ethereum informing
+   about the revealed deposit.
+5. Relayer bot submits the Wormhole VAA on Ethereum to the `SolanaDepositor`
+   contract.
+6. `SolanaDepositor` contract evaluates `depositID = keccak256(fundingTxHash | fundingOutputIndex)` 
+   and associates it with Solana depositor address.
+7. `SolanaDepositor` contract reveals the deposit to the `Bridge`
+8. tBTC is optimistically minted to `SolanaDepositor` contract address.
+9. Relayer bot notifies `SolanaDepositor` contract the deposit was
+   optimistically minted. `SolanaDepositor` inspects `TBTCVault.optimisticMintingRequests`
+   to confirm the amount and requests bridging tBTC from its own balance to the
+   Solana depositor address associated with the given deposit using Wormhole
+   Token Bridge. The same happens when the deposit was not optimistically minted
+   but swept.
+10. Relayer bot redeems the tBTC bridged to Solana to the depositor's Solana
```

Comment:
"redeems" is confusing to me. I link the redemption with a process of exiting tBTC to BTC. Is this a different kind of redemption?

```suggestion
10. Relayer bot transfers the tBTC bridged to Solana to the depositor's Solana
```

Reply:
Yes, it's Wormhole redemption as described [here](https://github.com/keep-network/tbtc-v2/pull/635#discussion_r1387711972). Please take a look at b1d234e98ffcc82c3a8e73257d3d8312eff70c6a. Hopefully, it clarifies it a bit.